src/citrine/builders/design_spaces.py

```diff
@@ -0,0 +1,205 @@
+try:
+    import pandas as pd
+except ImportError:  # pragma: no cover
+    raise ImportError('pandas>=0.25 is a requirement for the builders module')
+from itertools import product
+from typing import Mapping, Sequence, List
+from citrine.informatics.design_spaces import EnumeratedDesignSpace
+from citrine.informatics.descriptors import Descriptor, RealDescriptor
+
+
+def enumerate_cartesian_product(
+    design_grids: Mapping[str, Sequence],
+    descriptors: List[Descriptor],
+    name: str = 'Enumerated Cartesian product design space',
+    description: str = '',
+) -> EnumeratedDesignSpace:
+    """[ALPHA] Enumerate a Cartesian product from 1-D grids.
+
+    Given a dict of lists or tuples of values for each design descriptor,
+    create the list of candidates representing the Cartesian product of all
+    values for each descriptor
+
+    Parameters
+    ----------
+    design_grids: Mapping[str, Sequence]
+        {'descriptor key': [sequence, of, values]} for each descriptor
+    descriptors: List[Descriptor]
+        design descriptors representing the degrees of freedom of the design
+        space. descriptors' keys should match the keys of design_grids. If
+        none are passed, they will be auto-generated based on the keys
+        of design_grids.
+    name: str
+        name for the EnumeratedDesignSpace
+    description: str
+        description for the EnumeratedDesignSpace
+
+    """
+    design_space_tuples = list(product(*design_grids.values()))
+    design_space_cols = list(design_grids.keys())
+    df_ds = pd.DataFrame(data=design_space_tuples, columns=design_space_cols)
+    data = df_ds.to_dict('records')
+    design_space = EnumeratedDesignSpace(name=name,
+                                         description=description,
+                                         descriptors=descriptors,
+                                         data=data)
+    return design_space
+
+
+def enumerate_formulation_grid(
+    formulation_grids: Mapping[str, Sequence[float]],
```

Comment:
Singular, right?

Reply:
I think originally I was going for the idea that each dimension (each item in the dict) was its own grid, but grid kind of always implies more than one dimension. There's not a great word for a 1-D grid, is there? I agree that singular grid makes more sense.